src/support.cpp

```diff
@@ -48,6 +48,255 @@ bool AreaSupport::handleSupportModifierMesh(SliceDataStorage& storage, const Set
+    //
+    // In the example above, it shows a part of the model. For the gradual support infill, we process each layer in the following way:
+    //  -> Separate the generated support areas into isolated islands, and we process those islands one by one
+    //     so that the outlines and the gradual infill areas of an island can be printed together.
+    //
+    //  -> Calculate a number of layer groups, each group consists of layers that will be infilled with the same density.
+    //     The maximum number of densities is defined by the parameter "max graudual support steps". For example, it the <max steps> is 5,
+    //     It means that we can have at most 5 different densities, each is 1/2 than the other. So, it will looks like below:
+    //           |  step  |    density    |  description              |
+    //           |    0   |   1 / (2^0)   |  100% the most dense      |
+    //           |    1   |   1 / (2^1)   |                           |
+    //           |    2   |   1 / (2^2)   |                           |
+    //           |    3   |   1 / (2^3)   |                           |
+    //           |    4   |   1 / (2^4)   |                           |
+    //           |    5   |   1 / (2^5)   |  3.125% the least dense   |
+    //
+    //  -> With those layer groups, we can project areas with different densities onto the layer we are currently processing.
+    //     Like in the illustration above, you can see 4 different areas with densities ranging from 0 to 3.
+    //
+    //  -> We save those areas with different densities into the SupportInfillPart data structure, which holds all the support infill
+    //     information of a support island on a specific layer.
+    //
+    //  -> Note that this function only does the above, which is identifying and storing support infill areas with densities.
+    //     The actual printing part is done in FffGcodeWriter.
+    //
+
+    // no early-out for this function; it needs to initialize the [infill_area_per_combine_per_density]
+    float layer_skip_count = 8; // skip every so many layers as to ignore small gaps in the model making computation more easy
+    unsigned int gradual_support_step_layer_count = round_divide(gradual_support_step_height, storage.getSettingInMicrons("layer_height")); // The difference in layer count between consecutive density infill areas
+
+    // make gradual_support_step_height divisable by layer_skip_count
+    float n_skip_steps_per_gradual_step = std::max(1.0f, std::ceil(gradual_support_step_layer_count / layer_skip_count)); // only decrease layer_skip_count to make it a divisor of gradual_support_step_layer_count
+    layer_skip_count = gradual_support_step_layer_count / n_skip_steps_per_gradual_step;
+
+    size_t min_layer = 0;
+    size_t max_layer = total_layer_count - 1;
+
+    const coord_t wall_line_width_x = storage.getSettingInMicrons("wall_line_width_x");
+    const int skin_outline_count = 1;  // we only have 1 skin for support infill
```

Comment:
The name is a bit misleading here. This line count number is only used here to generate insets for a given infill area, and we only generate infill patterns within the insets, because otherwise, if the lines on the boundaries gets combined into a thicker line, it will be very wide and directly on top of the boundary, which can be too close to the model.

I have updated comment to explain this.